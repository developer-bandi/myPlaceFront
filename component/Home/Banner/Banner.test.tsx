import { act, renderHook } from "@testing-library/react";
import useBannerHook from "./BannerHook";

describe("Banner Hook 테스트", () => {
  jest.useFakeTimers();
  jest.spyOn(window, "setInterval");
  jest.spyOn(window, "clearInterval");

  it("setInterval test", () => {
    const { result } = renderHook(() => useBannerHook());
    act(() => {
      jest.advanceTimersByTime(8000);
    });
    expect(result.current.carouselNumber).toBe(1);
  });
});
