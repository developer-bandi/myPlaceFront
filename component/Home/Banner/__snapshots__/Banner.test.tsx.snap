// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Presentational 테스트 0번 배너 렌더링 1`] = `
<div>
  <section>
    <div
      class="mainBlock blue"
      data-testid="carousel0"
    >
      <div
        class="subBlock"
      >
        <div
          class="infoBlock"
        >
          <h3
            class="title"
          >
            자신의 취향에 맞는장소를
 MyPlace 에서 찾아보세요
          </h3>
          <p
            class="summary"
          >
            해시태그, 키워드를 활용하여
 원하는 장소를 검색해보세요
          </p>
        </div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27250%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="searchImg"
            data-nimg="intrinsic"
            decoding="async"
            src="undefined//slide/slide0_jlhmsf.svg"
            srcset="undefined//slide/slide0_jlhmsf.svg 1x, undefined//slide/slide0_jlhmsf.svg 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </div>
  </section>
</div>
`;

exports[`Banner Presentational 테스트 1번 배너 렌더링 1`] = `
<div>
  <section>
    <div
      class="mainBlock purple"
      data-testid="carousel1"
    >
      <div
        class="subBlock"
      >
        <div
          class="infoBlock"
        >
          <h3
            class="title"
          >
            장소에 관한 이야기를
 커뮤니티에서 나누어보세요
          </h3>
          <p
            class="summary"
          >
            다양한 사람들과
 장소에 관한 이야기를 나눌수 있습니다
          </p>
        </div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27250%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="searchImg"
            data-nimg="intrinsic"
            decoding="async"
            src="undefined//slide/slide1_ysk06a.svg"
            srcset="undefined//slide/slide1_ysk06a.svg 1x, undefined//slide/slide1_ysk06a.svg 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </div>
  </section>
</div>
`;

exports[`Banner Presentational 테스트 2번 배너 렌더링 1`] = `
<div>
  <section>
    <div
      class="mainBlock green"
      data-testid="carousel2"
    >
      <div
        class="subBlock"
      >
        <div
          class="infoBlock"
        >
          <h3
            class="title"
          >
            자신이 알고있는 장소를
 다른사람과 공유하기
          </h3>
          <p
            class="summary"
          >
            장소에 대한 정보를 제공하는것은
 사이트 발전에 도움이 됩니다
          </p>
        </div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27250%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="searchImg"
            data-nimg="intrinsic"
            decoding="async"
            src="undefined//slide/slide2_jckgbi.svg"
            srcset="undefined//slide/slide2_jckgbi.svg 1x, undefined//slide/slide2_jckgbi.svg 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </div>
  </section>
</div>
`;
